src/include/gurt/common.h

```diff
@@ -499,6 +499,38 @@ d_sgl_fini(d_sg_list_t *sgl, bool free_iovs)
 	sgl->sg_nr = 0;
 }
 
+static inline size_t
+d_sgl_buf_size(d_sg_list_t *sgl)
+{
+	size_t	size = 0;
+	int		i;
+
+	if (sgl == NULL || sgl->sg_iovs == NULL)
```

Comment:
For a new function is there a need to support sgl being NULL here?  An alternative would be to set `__attribute__((nonnull))` on the function and let the compiler work out the details. You need to keep the NULL check in daos_sgl_buf_size however for compatibility.

Reply:
ah, sure.